File: Contest/Contests/The 3rd Universal Cup Stage 0 Trial Contest/j.py
n, p = list(map(int, input().split()))
# print(n, p)
v = [0] * (n + 1)
# s = [0] * (n + 1)
for i in range(1, n + 1):
	x = int(input())
	v[i] = x
	# s[i] = s[i - 1] + x
# def f(x : int):
# 	temp = s[x - 1]
# 	temp += (x - 1) * (v[x] - 1)
# 	return p - temp < v[x]
# l, r = 0, n + 1
# while l + 1 != r:
# 	m = (l + r) // 2
# 	if f(m):
# 		r = m
# 	else:
# 		l = m
# print(l)
s = 0
ans = 0
for i in range(1, n + 1):
	ss = s + (v[i] - 1) * (i)
	if(ss >= p):
		# print(i)
		break;
	s += v[i]
	ans = i
print(ans)

File: Contest/Contests/The 3rd Universal Cup Stage 3 Ukraine/j.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long
#define all(x) x.begin(), x.end()

void sovle(){
    int n;
    std::cin >> n;
    std::vector<int> v(n + 1);
    for(int i = 1; i <= n; i++){
        std::cin >> v[i];
    }
    std::vector<int> vis(n + 1);
    std::vector<int> p;
    int flag = 0;
    for(int i = 1; i <= n; i++){
        if(!vis[v[i]]){
            int x = v[i], cnt = 0;
            std::vector<int> temp;
            while(!vis[x]){
                vis[x] = 1;
                x = v[x];       
                temp.push_back(x);
                cnt++;
            }
            if(temp.size() > p.size())p = temp;
            if(cnt == n){flag = 1;break;}
        }
    }
    if(flag){
        int ans = -1;
        int pos = 0;
        for(int i = 1; i <= n; i++){
            int t = std::min(v[i] - 1, i - 1) + std::min(n - v[i], n - i);
            if(t > ans){
                pos = i;
                ans = t;
            }
        }
        std::cout << ans << "\n";
        std::cout << "1\n" << pos << "\n";
    } else {
        std::cout << n << "\n";
        std::cout << p.size() << "\n";
        std::sort(all(p));
        for(auto i : p){
            std::cout << i << " ";
        }
        std::cout << "\n";
    }

}

signed main(){
    std::ios::sync_with_stdio(false),std::cin.tie(0),std::cout.tie(0);
    int _ = 1;
    std::cin >> _;
    while(_--){
        sovle();
    }
    return 0;
}